src/validator.cc

```diff
@@ -569,14 +570,42 @@ void Validator::CheckExpr(const Expr* expr) {
       break;
 
     case ExprType::Throw:
-      // TODO(karlschimpf) Define.
-      PrintError(&expr->loc, "Throw: don't know how to validate");
+      const Exception* except;
+      if (WABT_SUCCEEDED(CheckExceptVar(&expr->throw_.var, &except))) {
+        typechecker_.OnThrow(&except->sig);
+      }
       break;
 
-    case ExprType::TryBlock:
-      // TODO(karlschimpf) Define.
-      PrintError(&expr->loc, "TryBlock: don't know how to validate");
+    case ExprType::TryBlock: {
+      TryContext context;
```

Comment:
Ah, this is much clearer, I think. Thanks!

I think we'll eventually want to move more of this logic into TypeChecker, but this looks good for now.